spec/tasks/abs_spec.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'webmock/rspec'
+require_relative '../../tasks/abs.rb'
+require 'yaml'
+
+RSpec.shared_context('with_tmpdir') do
+  let(:tmpdir) { @tmpdir } # rubocop:disable RSpec/InstanceVariable
+
+  around(:each) do |example|
+    Dir.mktmpdir('rspec-provision_test') do |t|
+      @tmpdir = t
+      example.run
+    end
+  end
+end
+
+describe 'provision::abs' do
+  let(:abs) { ABSProvision.new }
+  let(:inventory_dir) { "#{tmpdir}/spec/fixtures" }
+  let(:inventory_file) { "#{inventory_dir}/litmus_inventory.yaml" }
+  let(:empty_inventory_yaml) do
+    <<~YAML
+      ---
+      version: 2
+      groups:
+      - name: docker_nodes
+        targets: []
+      - name: ssh_nodes
+        targets: []
+      - name: winrm_nodes
+        targets: []
+    YAML
+  end
+
+  include_context('with_tmpdir')
+
+  before(:each) do
+    FileUtils.mkdir_p(inventory_dir)
+  end
+
+  context '.run' do
+    it 'handles JSON parameters from stdin' do
+      json_input = '{"action":"foo","platform":"bar"}'
+      expect($stdin).to receive(:read).and_return(json_input)
+
+      expect { ABSProvision.run }.to(
+        raise_error(SystemExit) { |e|
+          expect(e.status).to eq(0)
+        }.and(
+          output("null\n").to_stdout,
+        ),
+      )
+    end
+
+    it 'raises an error when platform not given for provision' do
+      expect($stdin).to receive(:read).and_return('{"action":"provision"}')
+
+      expect { ABSProvision.run }.to raise_error(RuntimeError, %r{specify a platform when provisioning})
+    end
+
+    it 'raises an error when node_name not given for tear_down'
+    it 'raises an error if both node_name and platform are given'
+  end
+
+  context 'provision' do
+    let(:params) do
+      {
+        action: 'provision',
+        platform: 'redhat-8-x86_64',
+        inventory: tmpdir,
+      }
+    end
+    let(:response_body) do
+      [
+        {
+          'type'     => 'redhat-8-x86_64',
+          'hostname' => 'foo-bar.test',
+        },
+      ]
+    end
+
+    it 'provisions the platform' do
+      stub_request(:post, 'https://abs-prod.k8s.infracore.puppet.net/api/v2/request')
+        .to_return({ status: 202 }, { status: 200, body: response_body.to_json })
+
+      expect(abs.task(params)).to eq({ status: 'ok', nodes: 1 })
+
+      updated_inventory = YAML.load_file(inventory_file)
+      ssh_targets = updated_inventory['groups'].find { |g| g['name'] == 'ssh_nodes' }['targets']
+      expect(ssh_targets.size).to eq(1)
+      expect(ssh_targets.first.dig('facts', 'platform')).to eq('redhat-8-x86_64')
+    end
+
+    it 'provision with an existing inventory file' do
+      pending(<<~EOS)
+        XXX: It looks like there's an error hidden here in the way
```

Comment:
Should we file an issue on it? Issue tracker seems to be enabled.

Reply:
Sure: #186; I'll update the comment.